druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java

```diff
@@ -544,12 +557,15 @@ protected QuerySpec getQuery(RelDataType rowType, RexNode filter, List<RexNode>
         for (int groupKey : groupSet) {
           final String s = fieldNames.get(groupKey);
           if (s.equals(druidTable.timestampFieldName)) {
-            granularity = Granularity.NONE;
-            builder.add(s);
+            granularity = Granularity.ALL;
+            timeExtractionDimensionSpec = TimeExtractionDimensionSpec.makeFullTimeExtract();
+            builder.add(DruidConnectionImpl.DEFAULT_RESPONSE_TIMESTAMP_COLUMN);
+            dimensions.add(timeExtractionDimensionSpec);
+            //@TODO remove this assert cause we can have multiple extract on time column
```

Comment:
actually that todo is wrong, in case we are going a group by on time column implies that one that column will be used since it is not extraction.
in fact this is the case `select __time from table group by __time`